### PR TITLE
Introduce configurable timeout in TcpUtility

### PR DESCRIPTION
**Pull request by CyberDem0n**
the value is calculated based on the number of nodes and retry_timeout.

Close https://github.com/zalando/patroni/issues/2431

### CODE REVIEW

File: patroni/dcs/raft.py [hunk 1]
```diff
@@ -121,7 +121,8 @@ def __init__(self, on_ready, on_set, on_delete, **config):
                            journalFile=(file_template + '.journal' if self_addr else None),
                            onReady=on_ready, dynamicMembershipChange=True)
 
-        super(KVStoreTTL, self).__init__(self_addr, partner_addrs, conf)
+        retry_timeout = int(config.get('retry_timeout') or 10)
```
Comment (hughcapet) [inline patroni/dcs/raft.py:124]: Can `retry_timeout` be explicitly set to 0? We want it to be 10 then?

Comment (hughcapet) [reply patroni/dcs/raft.py:124]: Am also wondering: don't we want this code-wide default timeout = 10 be set via some variable?

**Comment by hughcapet** [comment]
👍

**Comment by idanovinda** [comment]
👍

**Pull request merged**